src/types/UiaTextRangeBase.cpp

```diff
@@ -1729,6 +1955,277 @@ UiaTextRangeBase::_moveEndpointByUnitCharacterBackward(gsl::not_null<IUiaData*>
     return std::make_tuple(start, end, degenerate);
 }
 
+// Routine Description:
+// - calculates new Endpoints/degenerate state if the indicated
+// endpoint was moved moveCount times by word.
+// Arguments:
+// - moveCount - the number of times to move
+// - endpoint - the endpoint to move
+// - moveState - values indicating the state of the console for the
+// move operation
+// - pAmountMoved - the number of times that the return values are "moved"
+// Return Value:
+// - A tuple of elements of the form <start, end, degenerate>
+std::tuple<Endpoint, Endpoint, bool> UiaTextRangeBase::_moveEndpointByUnitWord(gsl::not_null<IUiaData*> pData,
+                                                                               const int moveCount,
+                                                                               const TextPatternRangeEndpoint endpoint,
+                                                                               const MoveState moveState,
+                                                                               const std::wstring& wordDelimiters,
+                                                                               _Out_ gsl::not_null<int*> const pAmountMoved)
+{
+    if (moveState.Direction == MovementDirection::Forward)
+    {
+        return _moveEndpointByUnitWordForward(pData, moveCount, endpoint, moveState, wordDelimiters, pAmountMoved);
+    }
+    else
+    {
+        return _moveEndpointByUnitWordBackward(pData, moveCount, endpoint, moveState, wordDelimiters, pAmountMoved);
+    }
+}
+
+std::tuple<Endpoint, Endpoint, bool>
+UiaTextRangeBase::_moveEndpointByUnitWordForward(gsl::not_null<IUiaData*> pData,
+                                                 const int moveCount,
+                                                 const TextPatternRangeEndpoint endpoint,
+                                                 const MoveState moveState,
+                                                 const std::wstring& wordDelimiters,
+                                                 _Out_ gsl::not_null<int*> const pAmountMoved)
+{
+    *pAmountMoved = 0;
+    const int count = moveCount;
+    ScreenInfoRow currentScreenInfoRow = 0;
+    Column currentColumn = 0;
+
+    // set current location vars
+    if (endpoint == TextPatternRangeEndpoint::TextPatternRangeEndpoint_Start)
+    {
+        currentScreenInfoRow = moveState.StartScreenInfoRow;
+        currentColumn = moveState.StartColumn;
+    }
+    else
+    {
+        currentScreenInfoRow = moveState.EndScreenInfoRow;
+        currentColumn = moveState.EndColumn;
+    }
+
+    auto& buffer = pData->GetTextBuffer();
+    for (int i = 0; i < abs(count); ++i)

```

Comment:
This one has a lot of the same comments I made above. Is it a big copy/paste? Can it be deduplicated?

Reply:
making this a part of #3895
I'll get to this either this month or early next month.